packages/blocks/framework/src/lib/validators/validators.ts

```diff
@@ -221,6 +221,10 @@ export class Validators {
   static number: TypedValidatorFn<ValidationInputType.NUMBER> = {
     type: ValidationInputType.NUMBER,
     fn: (property, processedValue, userInput) => {
+      if (!property.required && (isNaN(userInput) || userInput.trim() === '')) {
```

Comment:
ah nice, had the same problem with cloudability and changed it to text property, this is better